bunsen-core/src/main/java/com/cerner/bunsen/definitions/LeafExtensionConverter.java

```diff
@@ -41,8 +41,27 @@ public void setField(IBase parentObject,
 
       fieldToSet.getMutator().addValue(parentObject, extension);
     }
+
+    /**
+     * Converts an object from a different data model to a HAPI object.
+     *
+     * @param input the object to convert
+     * @return the HAPI equivalent.
+     */
+    @Override
+    public IBase toHapi(Object input) {
+
+      IBase hapiObject = valuetoHapiConverter.toHapi(input);
```

Comment:
any object can be converted to Hapi or need to check for some type?

Reply:
Only objects implement HapiObjectConverter interface can call toHapi(Object input) to convert the input to a Hapi object.  The reason that we don't check the type of input Object here at L54 is because it is passing the input to the actual converter of an extension value field, for example an extension value can be a Premitive type of String or another composite type, and it continues to call toHapi(input) until it reaches the very leaf node of the converters where it will finally cast the input object to it's actual type.